EcpBtBot/DialogFlow/DialogFlowResponse.py

```diff
@@ -31,14 +31,17 @@ def fulfillment(self):
 
 class Metadata:
     def __init__(self,metadata):
-        self.intentId=metadata['intentId']
-        self.webhookUsed=metadata['webhookUsed']
-        self.webhookForSlotFillingUsed=metadata['webhookForSlotFillingUsed']
-        self.intentName=metadata['intentName']
+        self.intentName=""
+        if 'intentName' in metadata.keys():
```

Comment:
you can drop `.keys()` : `if 'intentName' in metadata` is enough.
Another thing you can do, a one-liner is: `self.intentName = metadata.get('intentName', '')`, meaning, if i find the key 'intentName' in the metadata dictionary, i get the value, and if not the default value is ''.

Reply:
ok

Reply:
ok